core/src/CMakeLists.txt

```diff
@@ -4,6 +4,22 @@ KOKKOS_INCLUDE_DIRECTORIES(
   ${KOKKOS_TOP_BUILD_DIR}
 )
 IF (Kokkos_ENABLE_IMPL_DESUL_ATOMICS AND NOT desul_FOUND)
```

Comment:
I think we should try to check for compatibility if an external `desul` installation is used. Ideally, the CMake variables used here would also become exported `desul` `CMake` options.

Reply:
I agree but this is beyond the scope of this PR

Reply:
I updated desul/desul#85 to export `desul_atomics_{CUDA,HIP,SYCL,OPENMP}` variables